agent/handlers/v1_handlers.go

```diff
@@ -82,14 +84,33 @@ func newTaskResponse(task *api.Task, containerMap map[string]*api.DockerContaine
 		desiredStatus = ""
 	}
 
-	return &v1.TaskResponse{
-		Arn:           task.Arn,
-		DesiredStatus: desiredStatus,
-		KnownStatus:   knownBackendStatus,
-		Family:        task.Family,
-		Version:       task.Version,
-		Containers:    containers,
+	resp.Arn = task.Arn
+	resp.DesiredStatus = desiredStatus
+	resp.KnownStatus = knownBackendStatus
+	resp.Family = task.Family
+	resp.Version = task.Version
+	resp.Containers = containers
+
+	return resp
+}
+
+func newContainerResponse(containerID string, container *api.DockerContainer, eni *api.ENI) v1.ContainerResponse {
+	resp := v1.ContainerResponse{}
+
+	resp.Name = containerID
+	resp.DockerId = container.DockerID
+	resp.DockerName = container.DockerName
+
+	if eni != nil {
```

Comment:
we should also populate `resp.Networks` when `eni == nil` , with whatever value we can get from container's metadata